tests/utils/test_basic_multicall.cairo

```diff
@@ -0,0 +1,182 @@
+fn test_no_data_for_multicall() {
+    // *********************************************************************************************
+    // *                              SETUP                                                        *
+    // *********************************************************************************************
+    let (role_store, data_store) = setup();
+
+    // *********************************************************************************************
+    // *                              TEST LOGIC                                                   *
+    // *********************************************************************************************
+
+    let mut calls = array![];
+    let mut calldata_param = array![1, 42];
+    let first_call = Call {
+        to: data_store.contract_address,
+        selector: selector!("set_felt25"), /// generate keccak hash for 'set_felt252' in cairo
+        calldata: calldata_param
+    };
+
+    // should panic due to empty calls. Notice that calls has no append()
+    let result: Array<Span<felt252>> = multicall(calls);
+
+    // check first call result
+    assert(data_store.get_felt252(1).unwrap() == 42, 'Invalid value');
+
+    // *********************************************************************************************
+    // *                              TEARDOWN                                                     *
+    // *********************************************************************************************
+    teardown(data_store.contract_address);
+}
+
+/// Utility function to deploy a data store contract and return its address.
+///
+/// # Arguments
+///
+/// * `role_store_address` - The address of the role store contract.
+///
+/// # Returns
+///
+/// * `ContractAddress` - The address of the deployed data store contract.
+fn deploy_data_store(role_store_address: ContractAddress) -> ContractAddress {
+    let contract = declare('DataStore');
+    let mut constructor_calldata = array![];
+    constructor_calldata.append(role_store_address.into());
+    contract.deploy(@constructor_calldata).unwrap()
+}
+
+/// Utility function to deploy a role store contract and return its address.
+///
+/// # Returns
+///
+/// * `ContractAddress` - The address of the deployed role store contract.
+fn deploy_role_store() -> ContractAddress {
+    let contract = declare('RoleStore');
+    let constructor_arguments: @Array::<felt252> = @ArrayTrait::new();
+    contract.deploy(constructor_arguments).unwrap()
+}
+
+/// Utility function to setup the test environment.
+///
+/// # Returns
+///
+/// * `ContractAddress` - The address of the caller.
+/// * `IRoleStoreSafeDispatcher` - The role store dispatcher.
+/// * `IDataStoreSafeDispatcher` - The data store dispatcher.
+/// * `ContractAddress` - The role store contract address.
+/// * `ContractAddress` - The data store contract address.
+fn setup() -> (IRoleStoreSafeDispatcher, IDataStoreSafeDispatcher) {
```

Comment:
you could import setup like here to avoid copy pasta https://github.com/keep-starknet-strange/satoru/blob/main/tests/data/test_data_store.cairo#L6

Reply:
Sure thank you !